CoreUtilities/test/unit/src/au/gov/asd/tac/constellation/utilities/gui/JDropDownMenuNGTest.java

```diff
@@ -0,0 +1,235 @@
+import static org.testng.Assert.assertEqualsNoOrder;
+import static org.testng.Assert.assertNotNull;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+/**
+ *
+ * @author Auriga2
+ */
+public class JDropDownMenuNGTest {
+
+    private List<String> items = new ArrayList<String>();
+    private static MockedStatic<InstalledFileLocator> installedFileLocatorMockedStatic;
+    private static InstalledFileLocator installedFileLocatorMocked;
+
+    private static final String COOKIE_ICON_PATH = "au/gov/asd/tac/constellation/utilities/modules/ext/icons/cookie.png";
+    private static final String DROP_DOWN_ARROW_ICON_PATH = "au/gov/asd/tac/constellation/utilities/modules/ext/icons/drop_down_arrow.png";
+    private static final File iconFile = new File(DROP_DOWN_ARROW_ICON_PATH);
+
+    @BeforeClass
+    public static void setUpClass() throws Exception {
+        installedFileLocatorMocked = mock(InstalledFileLocator.class);
+        installedFileLocatorMockedStatic = Mockito.mockStatic(InstalledFileLocator.class);
+    }
+
+    @AfterClass
+    public static void tearDownClass() throws Exception {
+        installedFileLocatorMockedStatic.close();
+    }
+
+    @BeforeMethod
+    public void setUpMethod() throws Exception {
+        installedFileLocatorMockedStatic.when(InstalledFileLocator::getDefault)
+                .thenReturn(installedFileLocatorMocked);
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault()
+                .locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false))
+                .thenReturn(iconFile);
+        items.clear();
+        items.add("Item 1");
+        items.add("Item 2");
+        items.add("Item 3");
+        items.add("Item 4");
+    }
+
+    @AfterMethod
+    public void tearDownMethod() throws Exception {
+        installedFileLocatorMockedStatic.reset();
+    }
+
+    /**
+     * Test the constructor works with a Text
+     */
+    @Test
+    public void testConstructor_WithText() {
+        final File iconFile = new File("non_existing_icon.png");
+        installedFileLocatorMockedStatic.when(() -> InstalledFileLocator.getDefault().locate("modules/ext/icons/drop_down_arrow.png", "au.gov.asd.tac.constellation.utilities", false)).thenReturn(iconFile);
+
+        JDropDownMenu instance = new JDropDownMenu("Text", items);
+        String expResult = "Text";
+
+        String result = instance.getText();
+        assertEquals(result, expResult);
+    }
+
+    /**
+     * Test the constructor works with an Icon
+     */
+    @Test
+    public void testConstructor_WithIcon() {
+        final Icon icon = new ImageIcon(COOKIE_ICON_PATH);
+        JDropDownMenu instance = new JDropDownMenu(icon, items);
+        Icon result = instance.getIcon();
+        assertNotNull(result);
```

Comment:
Would you be able to compare that the icon is the same?
ie. `assertEquals(result, icon);`

Reply:
No they were `CompoundIcon`s and I couldn't compare them to be equal.

Reply:
You could potentially check if the Icon added is contained in the CompoundIcon (casting `result` as `CompoundIcon`)?

Reply:
Yes it worked. Thanks Josh.